app/src/main/java/com/teamltt/carcare/activity/LoginActivity.java

```diff
@@ -0,0 +1,216 @@
+        }
+    }
+
+    private void handleSignInResult(GoogleSignInResult result) {
+        Log.d(TAG, "handleSignInResult:" + result.isSuccess());
+        if (result.isSuccess()) {
+            // Signed in successfully, show authenticated UI.
+            googleSignInAccount = result.getSignInAccount();
+            tvStatus.setText(getString(R.string.signed_in_fmt, googleSignInAccount.getDisplayName()));
+            updateUI(true);
+        } else {
+            // Signed out, show unauthenticated UI.
+            updateUI(false);
+        }
+    }
+
+    private void signIn() {
+        Intent signInIntent = Auth.GoogleSignInApi.getSignInIntent(google_api_client);
+        startActivityForResult(signInIntent, RC_SIGN_IN);
+    }
+
+    @Override
+    public void onConnectionFailed(ConnectionResult connectionResult) {
+        // An unresolvable error has occurred and Google APIs (including Sign-In) will not
+        // be available.
+        Log.d(TAG, "onConnectionFailed:" + connectionResult);
+    }
+
+
+    private void showProgressDialog() {
+        if (progress_dialog == null) {
+            progress_dialog = new ProgressDialog(this);
+            progress_dialog.setMessage(getString(R.string.loading));
+            progress_dialog.setIndeterminate(true);
+        }
+
+        progress_dialog.show();
+    }
+
+
+    private void hideProgressDialog() {
+        if (progress_dialog != null && progress_dialog.isShowing()) {
+            progress_dialog.hide();
+        }
+    }
+
+
+    private void updateUI(boolean signedIn) {
+        if (signedIn) {
+            String firstName = googleSignInAccount.getGivenName();
+            String lastName = googleSignInAccount.getFamilyName();
+            String google_id = googleSignInAccount.getId();
+
+            // Add user to database
+            dbHelper = new DbHelper(this);
+            // TODO Make sure you're not adding duplicates to the database
+            if (!dbHelper.containsUser(google_id)) {
+                dbHelper.createNewUser(google_id, firstName, lastName);
+            }
+
+            // Go to Home Screen
+            Intent intent = new Intent(this, HomeActivity.class);
+            intent.putExtra(EXTRA_MESSAGE + ".FIRSTNAME", firstName);
+            intent.putExtra(EXTRA_MESSAGE + ".LASTNAME", lastName);
+            intent.putExtra(EXTRA_MESSAGE + ".USERID", google_id);
+        } else {
+            tvStatus.setText(R.string.please_sign_in);
+
+            findViewById(R.id.btnGoogleSignIn).setVisibility(View.VISIBLE);
+        }
+    }
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()) {
+            case R.id.btnGoogleSignIn:
+                signIn();
+                break;
+        }
+    }
+
+    @Override
+    protected void onStop() {
+        super.onStop();
+        if (google_api_client.isConnected()) {
+            google_api_client.disconnect();
+        }
+    }
+
+    @Override
+    protected void onResume() {
```

Comment:
If the user navigates away from the app to the home screen, the activity is stopped and the client is disconnected. When they navigate back, a login is attempted in `onStart` with a disconnected client. Before we attempt a login, we should ensure that the client is connected, or connect if it isn't.